src/publish/posit-cloud/posit-cloud.ts

```diff
@@ -0,0 +1,304 @@
+import {
+  readAccessTokens,
+  writeAccessToken,
+  writeAccessTokens,
+} from "../common/account.ts";
+
+import { createTempContext } from "../../core/temp.ts";
+import { completeMessage, withSpinner } from "../../core/console.ts";
+import { RenderFlags } from "../../command/render/types.ts";
+import { createBundle } from "../common/bundle.ts";
+import { md5HashBytes } from "../../core/hash.ts";
+
+export const kPositCloud = "posit-cloud";
+const kPositCloudDescription = "Posit Cloud";
+
+export const kPositCloudServerVar = "POSIT_CLOUD_SERVER";
+export const kPositCloudTokenVar = "POSIT_CLOUD_TOKEN";
+export const kPositCloudAuthTokenSecretVar = "POSIT_CLOUD_SECRET";
+
+export const positCloudProvider: PublishProvider = {
+  name: kPositCloud,
+  description: kPositCloudDescription,
+  requiresServer: true,
+  listOriginOnly: true,
+  accountTokens,
+  authorizeToken,
+  removeToken,
+  resolveTarget,
+  publish,
+  isUnauthorized,
+  isNotFound,
+};
+
+type Account = {
+  username: string;
+  token: string;
+  tokenSecret: string;
+};
+
+function getServer(): string {
+  return Deno.env.get(kPositCloudServerVar) || "https://api.posit.cloud";
+}
+
+function createClientFromAccountToken(
+  accountToken: AccountToken,
+): PositCloudClient {
+  const [token, tokenSecret] = accountToken.token.split("|");
+  return new PositCloudClient(getServer(), token, tokenSecret);
+}
+
+function combineTokenAndSecret(token: string, tokenSecret: string): string {
+  return `${token}|${tokenSecret}`;
+}
+
+function accountTokens() {
+  const accounts: AccountToken[] = [];
+
+  // check for environment variable
+  const server = Deno.env.get(kPositCloudServerVar);
+  const token = Deno.env.get(kPositCloudTokenVar);
+  const tokenSecret = Deno.env.get(kPositCloudAuthTokenSecretVar);
+  if (server && token && tokenSecret) {
+    accounts.push({
+      type: AccountTokenType.Environment,
+      name: kPositCloudTokenVar,
+      server,
+      token: combineTokenAndSecret(token, tokenSecret),
+    });
+  }
+
+  // check for recorded tokens
+  const tokens = readAccessTokens<Account>(kPositCloud);
+  if (tokens) {
+    accounts.push(...tokens.map((token) => ({
+      type: AccountTokenType.Authorized,
+      name: token.username,
+      server: null,
+      token: combineTokenAndSecret(token.token, token.tokenSecret),
+    })));
+  }
+
+  return Promise.resolve(accounts);
+}
+
+function removeToken(token: AccountToken) {
+  writeAccessTokens(
+    positCloudProvider.name,
+    readAccessTokens<Account>(positCloudProvider.name)?.filter(
+      (accessToken) => {
+        return accessToken.username !== token.name;
+      },
+    ) || [],
+  );
+}
+
+async function authorizeToken(
+  _options: PublishOptions,
+  _target?: PublishRecord,
+): Promise<AccountToken | undefined> {
+  // get credentials
+  while (true) {
+    const token = await Input.prompt({
+      message: "Token:",
+      hint: "Learn more at https://posit.cloud/learn/guide#account-tokens",
```

Comment:
I think here and in the subsequent hint you should be directing the user to the exact page where they provision a token and secret for their account. With this link, they get once sentence about tokens, then need to realize that they should click "Publish Outputs", then read some general conceptual material about Publish Outputs, then realize that they need to skip past "Publish from a Cloud Project", then ???. I think this link should land them directly on the provisioning page and if additional explanation is required it should be there inline.

Reply:
Posit Cloud doesn't have such a URL at the moment, unfortunately... The URL for that page is specific to the user's account. I filed an issue for us to add a generic tokens URL and took out the `hint`.